ShelterModule/Program.cs

```diff
@@ -20,6 +21,10 @@ public static void Main(string[] args)
         builder.Services.AddEndpointsApiExplorer();
         builder.Services.AddSwaggerGen();
 
+        var keyVaultUrl = new Uri(builder.Configuration.GetSection("KeyVaultURL").Value!);
+        var azureCredential = new DefaultAzureCredential();
+        builder.Configuration.AddAzureKeyVault(keyVaultUrl, azureCredential);
+
```

Comment:
I think we could move this to `ConfigureOptions()` since we add a config source here. Also, we probably shouldn't do this if `app.Environment.IsDevelopment()`, because it makes it impossible to run the app locally. 

Also we can use `GetValue()` here as well (like below).

Reply:
BTW this is the reason why tests fail - we get an exception from `new DefaultAzureCredential()`.